sklearn/preprocessing/_encoders.py

```diff
@@ -770,8 +782,15 @@ def fit(self, X, y=None):
         """
         # base classes uses _categories to deal with deprecations in
         # OneHoteEncoder: can be removed once deprecations are removed
+        _set_config(assume_finite=True)
```

Comment:
Firstly, this configuration will remain set after fit is finished, secondly it will incorrectly pass through inf without error, thirdly it is not thread safe. We should instead use force_all_finite in check_array.

Reply:
I see.. definitely not a good option then. I added a `force_all_finite` passed to `_check_X` to give optionality for `NaN`'s.